src/oc_span_sweeper.erl

```diff
@@ -0,0 +1,99 @@
+%%%------------------------------------------------------------------------
+%% Copyright 2018, OpenCensus Authors
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%%------------------------------------------------------------------------
+-module(oc_span_sweeper).
+
+-behaviour(gen_statem).
```

Comment:
Why `gen_statem` instead of `gen_server`?

Reply:
Just because it has `state_timeout`. With a `gen_server` to achieve similar you have to manually do an `erlang:start_timer/3`, track the ref and all that, since the only timeout option is based on if a message is received, so annoyingly gets canceled on any message (not just the ones you care about).